drivers/accel/adxl367/iio_adxl367.c

```diff
@@ -0,0 +1,936 @@
+ * @brief Reads the number of given samples for the selected channels
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param samples - Number of samples to be returned
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl367_iio_read_samples(void* dev, int* buff, uint32_t samples)
+{
+	int16_t data_x;
+	int16_t data_y;
+	int16_t data_z;
+	int16_t raw_temp;
+	struct adxl367_iio_dev *iio_adxl367;
+	struct adxl367_dev *adxl367;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl367 = (struct adxl367_iio_dev *)dev;
+
+	if (!iio_adxl367->adxl367_dev)
+		return -EINVAL;
+
+	adxl367 = iio_adxl367->adxl367_dev;
+
+	for(uint32_t i = 0; i < samples*iio_adxl367->no_of_active_channels;) {
+		adxl367_get_raw_xyz(adxl367, &data_x, &data_y, &data_z);
+
+		if (iio_adxl367->active_channels & NO_OS_BIT(0)) {
+			buff[i] = no_os_sign_extend32(data_x, 15);
+			i++;
+		}
+		if (iio_adxl367->active_channels & NO_OS_BIT(1)) {
+			buff[i] = no_os_sign_extend32(data_y, 15);
+			i++;
+		}
+		if (iio_adxl367->active_channels & NO_OS_BIT(2)) {
+			buff[i] = no_os_sign_extend32(data_z, 15);
+			i++;
+		}
+		if (iio_adxl367->active_channels & NO_OS_BIT(3)) {
+			adxl367_read_raw_temp(adxl367, &raw_temp);
+			buff[i] = raw_temp;
+			i++;
+		}
+	}
+
+	return samples;
+}
+
+/***************************************************************************//**
+ * @brief Initializes the ADXL367 IIO driver
+ *
+ * @param iio_dev    - The iio device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       		   parameters.
+ *
+ * @return ret       - Result of the initialization procedure.
+*******************************************************************************/
+int adxl367_iio_init(struct adxl367_iio_dev **iio_dev,
+		     struct adxl367_iio_init_param *init_param)
+{
+	int ret;
+	struct adxl367_iio_dev *desc;
+
+	desc = (struct adxl367_iio_dev *)calloc(1, sizeof(*desc));
+	if (!desc)
+		return -ENOMEM;
+
+	desc->iio_dev = &adxl367_iio_dev;
+
+	// Initialize ADXL367 driver
+	ret = adxl367_init(&desc->adxl367_dev, *(init_param->adxl367_initial_param));
+	if (ret)
+		goto error_adxl367_init;
+
+	// Enable temperature reading
+	ret = adxl367_temp_read_en(desc->adxl367_dev, 1);
+	if (ret)
+		goto error_adxl367_init;
+
+	// Set ODR to 400Hz for iio live data feed
+	ret = adxl367_set_output_rate(desc->adxl367_dev, ADXL367_ODR_400HZ);
+	if (ret)
+		goto error_adxl367_init;
+
+	// Enter measure mode
+	ret = adxl367_set_power_mode(desc->adxl367_dev, ADXL367_OP_MEASURE);
+	if (ret)
+		goto error_adxl367_init;
+
+	*iio_dev = desc;
+
+	return 0;
+
+error_adxl367_init:
+	free(desc);
+	return ret;
+error_config:
+	adxl367_remove(desc->adxl367_dev);
+	free(desc);
+	return ret;
```

Comment:
it doesn't look like this label is used. can it be removed?

Reply:
If one function fails, ret is used. 

ret = adxl367_init(&desc->adxl367_dev, *(init_param->adxl367_initial_param));
	if (ret)
		goto error_adxl367_init;

error_adxl367_init:
	free(desc);
	return ret;